src/controllers/keyboard/keyboardcontrollerpreset.cpp

```diff
@@ -0,0 +1,242 @@
+            // due to multiple keys sharing the same character)
+            QList<int> scancodes = layoutUtils::findScancodesForCharacter(originalKeyseqLayout, key.at(0), modifier);
+
+            // Warn user if any scancodes where found for this character
+            if (scancodes.isEmpty()) {
+                qWarning() << "Couldn't find any scancodes for character"
+                           << key.at(0) << "on" << keyseqsRawI->byPositionOf
+                           << (onlyShift ? "with" : "without") << "shift."
+                           << "Skipping mapping for" << configKey;
+                continue;
+            }
+
+            // Remove original key sequence
+            keyseqs.clear();
+
+            // Iterate through all found scancodes
+            for (const int& scancode : scancodes) {
+
+                // Get KbdKeyChar
+                const KbdKeyChar *keyChar = layoutUtils::getKbdKeyChar(layout, (unsigned char) scancode, modifier);
+                QChar character(keyChar->character);
+
+                // If key is not dead, reconstruct key sequence with translated character. Otherwise,
+                // warn user about key that won't work.
+                if (!keyChar->isDead) {
+                    QString modifiersString = modifiers.join("+");
+                    if (!modifiersString.isEmpty()) {
+                        modifiersString += "+";
+                    }
+                    keyseqs.append(modifiersString + character);
+                } else {
+                    qWarning() << "Can't use key with scancode " << scancode
+                               << " because it's a dead key on layout '" << layoutName
+                               << "'. Please specify <keyseq lang=\"" << layoutName
+                               << "\" scancode=\"" << scancode << "\"> in the keyboard preset file (group: "
+                               << configKey.group << ").";
+                    keyseqs[0] = "";
+                }
+            }
+        }
+
+        for (QString& keyseq : keyseqs) {
+            QString key = layoutUtils::getCharFromKeysequence(keyseq);
+            QStringList mods = layoutUtils::getModifiersFromKeysequence(keyseq);
+
+            // Clear keyseq
+            keyseq = "";
+
+            // NOTE: This order must be the same as in
+            // ...   KeyboardEventFilter::getKeySeq()
+            if (mods.contains("Shift")) keyseq += "Shift+";
+            if (mods.contains("Ctrl")) keyseq += "Ctrl+";
+            if (mods.contains("Alt")) keyseq += "Alt+";
+            if (mods.contains("Meta")) keyseq += "Meta+";
+
+            // If the key is modified, the comparisons in KeyboardEventFilter
+            // are based on QKeySequence::toString()
+            if (!keyseq.isEmpty()) {
+              key = QKeySequence(key).toString();
+            }
+
+            // Reconstruct key sequence with ordened modifiers
+            keyseq += key;
```

Comment:
I propose to provide functions for concatenating key sequences. At this point it is not immediately obvious that you already prepended "+" some lines before and what happens if key is empty. Using generic string operations is error prone.